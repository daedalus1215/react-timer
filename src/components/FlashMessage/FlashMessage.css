.flash-message,
.flash-message:hover {
    padding: 1em 1em 1em 2em;
    margin-bottom: .25em;
    background:#5ec232;
    color:white;
    font-size: 1.1rem;
    cursor: pointer;
    display: flex;
}

.flash-message_message {
    flex: 1 1;
}

.flash-message:hover {
    background: #418823;
}

.flash-message_cancel {
    display: inline-block;
    
}