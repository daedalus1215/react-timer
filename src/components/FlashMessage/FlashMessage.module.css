.flashMessage,
.flashMessage:hover {
    position: absolute;
    top: 50px;
    right: 50px;
    display: flex;
    background:#5ec232;
    color:white;
    font-size: 1.1rem;
    cursor: pointer;
    width: 200px;
}

.flashMessage:hover {
    background: #418823;
}

.flashMessageContent { 
    flex: 1 1;
    padding: 10px;
    font-size: 12px;
}

.flashMessageCancel {
    flex: 1 1;
    text-align: right;
    font-size: 14px;
    padding: 10px;
}