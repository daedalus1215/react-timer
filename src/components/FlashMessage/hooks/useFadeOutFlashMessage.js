import { useEffect } from 'react';
import useFlashMessageContext from 'hooks/useFlashMessageContext';


const useFadeOutFlashMessage = () => {
    const { message, error, resetFlashMessage } = useFlashMessageContext();

    useEffect(() => {
        const timer = !error && setTimeout(() => resetFlashMessage(), 10000);
        
        return () => clearTimeout(timer);
    }, [message, resetFlashMessage, error]);
}

export default useFadeOutFlashMessage;