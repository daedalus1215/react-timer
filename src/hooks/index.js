export { default as useFetchProjectOptions } from './useFetchProjectOptions';
export { default as useGetProjectOptionLabel } from './useGetProjectOptionLabel';
export { default as useTaskEditContext } from './useTaskEditContext';
export { default as useBackButtonListener } from './useBackButtonListener';
export { default as useFetchTaskById } from './useFetchTaskById';
export { default as useLoadinSpinnerContext } from './useLoadinSpinnerContext';
export { default as useFlashMessageContext } from './useFlashMessageContext';