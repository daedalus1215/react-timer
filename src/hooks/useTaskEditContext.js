import { useContext } from 'react';
import TaskEditFormContext from '../contexts/TaskEditFormContext';

const useTaskEditContext = () => useContext(TaskEditFormContext);

export default useTaskEditContext;
