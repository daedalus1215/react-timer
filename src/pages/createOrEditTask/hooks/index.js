export { default as useFetchTags } from './useFetchTags';
export { default as useTagContext } from './useTagContext';
export { default as useTagTransformer } from './useTagTransformer';
export { default as useFetchTaskById } from './useFetchTaskById/useFetchTaskById';