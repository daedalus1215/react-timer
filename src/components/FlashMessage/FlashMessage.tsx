import React from 'react';
import cn from 'classnames';
import { useDispatch } from 'react-redux';
import { TEN_SECONDS } from 'utils/constants';
import useTimeout from 'hooks/useTimeout';
import { setFlashMessageAction } from 'redux/actionCreators/actions';
import { useFlashMessageSelectors } from 'redux/selectors/useFlashMessageSelectors';
import './FlashMessage.css';

const FlashMessage = () => {
  const { message, status } = useFlashMessageSelectors();
  const isSuccess = status === 'success';
  const isInfo = status === 'info';
  const isError = status === 'error';
  const dispatch = useDispatch();

  useTimeout(() => {
    dispatch(setFlashMessageAction('success', null));
  }, TEN_SECONDS, message);


  return (
    !message || (
      <div
        className={cn('flashMessage', {
          'fade-out': !status.error,
          success: isSuccess,
          info: isInfo,
          error: isError,
        })}
        data-testid="flash-message"
        onClick={() => dispatch(setFlashMessageAction('success', null))}
      >
        <div
          className={cn('flashMessageIcon', {
            'glyphicon glyphicon-ok-sign': isSuccess,
            'glyphicon glyphicon-exclamation-sign': isError,
            'glyphicon glyphicon-info-sign': isInfo,
          })}
        />

        <div className="flashMessageContent">{message}</div>

        <div className={cn('flashMessageCancel', 'glyphicon glyphicon-remove-circle')} />
      </div>
    )
  );
};

export default FlashMessage;
