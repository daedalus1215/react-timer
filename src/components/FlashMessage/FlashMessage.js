import React from 'react';
import PropTypes from 'prop-types';
import useTaskEditContext from 'hooks/useTaskEditContext';
import './FlashMessage.css';

const FlashMessage = () => {
  const { message, setMessage } = useTaskEditContext();

  return (
    !message || (
      <div className="flash-message" onClick={() => setMessage(null)}>
        <div className="flash-message_message" test-data-id="flash-message">{message}</div>
        <div className="flash-message_cancel">X</div>
      </div>
    )
  );
};

FlashMessage.propTypes = {
  message: PropTypes.string,
  onClick: PropTypes.func
};

export default FlashMessage;
