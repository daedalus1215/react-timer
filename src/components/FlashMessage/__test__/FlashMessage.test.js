import React from 'react';
import sinon from 'sinon';
import chai, { expect } from 'chai';
import SinonChai from 'sinon-chai';
import FlashMessage from '../FlashMessage';
import * as useTaskEditContext from 'pages/CreateOrEditTask/Form/EditTask/hooks/useTaskEditContext';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
chai.use(SinonChai);

//@TODO: Test the clicking functionality
describe('src/components/FlashMessage/__test__/FlashMessage.test.js', () => {
  describe('FlashMessage', () => {
    let wrapper;
    let context;
    const stuber = sinon.stub(useTaskEditContext, 'default');

    it('should not show FlashMessage when no message is present.', () => {
      context = {
        message: undefined
      };
      stuber.returns(context);

      wrapper = mount(<FlashMessage />);
      expect(wrapper).to.be.empty;
    });

    it('should show FlashMessage when message has a value.', () => {
      context = {
        message: 'Success'
      };
      stuber.returns(context);

      wrapper = mount(<FlashMessage />);
      expect(wrapper.find("[test-data-id='flash-message']").text()).to.be.eql(
        'Success'
      );
    });
  });
});
