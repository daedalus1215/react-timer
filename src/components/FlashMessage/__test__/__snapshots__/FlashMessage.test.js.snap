// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/FlashMessage/__test__/FlashMessage.test.js FlashMessage should not show FlashMessage when 'message' is NOT present. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flashMessage fade-out success"
        data-testid="flash-message"
      >
        <div
          class="flashMessageIcon glyphicon glyphicon-ok-sign"
        />
        <div
          class="flashMessageContent"
        >
          this message should make the component show, since we have a message
        </div>
        <div
          class="flashMessageCancel glyphicon glyphicon-remove-circle"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flashMessage fade-out success"
      data-testid="flash-message"
    >
      <div
        class="flashMessageIcon glyphicon glyphicon-ok-sign"
      />
      <div
        class="flashMessageContent"
      >
        this message should make the component show, since we have a message
      </div>
      <div
        class="flashMessageCancel glyphicon glyphicon-remove-circle"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
